### ISSUE TITLE
Reorg of variables based on categories

### ISSUE THREAD

**Issue by manmeetkaur** [astronomer/docs#2682]
### Issue Type

Missing content for an existing feature

### Links to Affected Docs

https://docs.astronomer.io/astro/platform-variables

### The Issue or Context

This is a follow-up issue for the issue #2623, and the [related conversation](https://github.com/astronomer/docs/pull/2623#discussion_r1280016534).

- To a user it seems like all env variables can be set from the Cloud UI or Dockerfile.
- We have no checks in the Cloud UI to restrict user from setting certain variables
- We have not published any reserved keywords, which should not be used by users, as indicated by @melugoyal 's [comment](https://github.com/astronomer/docs/pull/2623#discussion_r1280015579)

### Fix or Proposed Change

In a nutshell:

- call out the list of env vars, that seem like can be overwritten but actually can't be. using `info`
- List the variables that can be overwritten and can have an adverse impact
- list variables that we treat as "reserved" so that accidentally users should not be creating those. 

@cmarteepants / @paolaperaza Do we intend to incorporate checks within the UI to prevent users from overriding or utilizing certain variables as part of H2 product plan?
